### PR TITLE
Systemd: cosmetic modifications

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Description of changes

Trying to do https://github.com/NixOS/nixpkgs/pull/269620 the right way.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by AndersonTorres** [comment]
With https://github.com/NixOS/nixpkgs/pull/271271 merged, now it can be included in systemd

**Comment by l0b0** [comment]
Love the refactors, thank you!

**Review by flokli — APPROVED** [review]

**Comment by nikstur** [comment]
The only reason I'm kind of against this is that https://github.com/NixOS/nixpkgs/pull/265951 will change much of this again. For example, many of the `mesonBool` options will have to be converted to `mesonEnable`. It seems like a lot of churn to rebase the systemd-255 PR onto these changes. It's probably easier to first merge systemd 255 and then do this refactor.

**Comment by AndersonTorres** [comment]
Well, you are certainly free to "deprecate" this PR by "cherry-picking" the modification.

It is not so hard to convert the `-D<feature>=<value>` strings to `lib.mesonOption feature value` and friends.

Indeed it makes the things way easier in the future: changing a `mesonBool` to `mesonEnable` is easier than changing a cryptic `=true` to `=enabled`.

**Comment by ElvishJerricco** [comment]
I'm inclined to agree with @nikstur and wait for https://github.com/NixOS/nixpkgs/pull/265951 to be merged.

File: pkgs/os-specific/linux/systemd/default.nix [hunk 1]
```diff
@@ -820,15 +820,31 @@ stdenv.mkDerivation (finalAttrs: {
...
+      supports SysV and LSB init scripts and works as a replacement for
+      sysvinit. Other parts include a logging daemon, utilities to control basic
+      system configuration like the hostname, date, locale, maintain a list of
+      logged-in users and running containers and virtual machines, system
+      accounts, runtime directories and settings, and daemons to manage simple
+      network configuration, network time synchronization, log forwarding, and
+      name resolution.
+    '';
+    license = with lib.licenses; [ lgpl21Plus ];
+    maintainers = with lib.maintainers; [ flokli kloenk ];
```
Comment (RaitoBezarius) [inline pkgs/os-specific/linux/systemd/default.nix:843]: shouldn't the maintainers be the systemd team here?

File: pkgs/os-specific/linux/systemd/default.nix [hunk 2]
```diff
@@ -820,15 +820,31 @@ stdenv.mkDerivation (finalAttrs: {
...
+      elaborate transactional dependency-based service control logic. systemd
+      supports SysV and LSB init scripts and works as a replacement for
+      sysvinit. Other parts include a logging daemon, utilities to control basic
+      system configuration like the hostname, date, locale, maintain a list of
+      logged-in users and running containers and virtual machines, system
+      accounts, runtime directories and settings, and daemons to manage simple
+      network configuration, network time synchronization, log forwarding, and
+      name resolution.
+    '';
+    license = with lib.licenses; [ lgpl21Plus ];
```
Comment (RaitoBezarius) [inline pkgs/os-specific/linux/systemd/default.nix:842]: isn't this wrong?
https://github.com/systemd/systemd/commit/5430f7f2bc7330f3088b894166bf3524a067e3d8
mentions MIT bits and GPL2+ bits.

File: pkgs/os-specific/linux/systemd/default.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/os-specific/linux/systemd/default.nix:843]: Dunno, it was there when I touched it.

File: pkgs/os-specific/linux/systemd/default.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/os-specific/linux/systemd/default.nix:842]: Dunno, it was there when I touched it (2).

Comment (AndersonTorres) [reply pkgs/os-specific/linux/systemd/default.nix:842]: Main site removed MIT: https://github.com/systemd/systemd/blob/main/LICENSE.MIT

Comment (flokli) [reply pkgs/os-specific/linux/systemd/default.nix:842]: Check https://github.com/systemd/systemd/blob/720c618397397f958caeb050a1528eb0d6f7a4a6/LICENSES/README.md?plain=1#L37, it explains the state of licensing.

Comment (AndersonTorres) [reply pkgs/os-specific/linux/systemd/default.nix]: Updated.

**Review by RaitoBezarius — DISMISSED** [review]
Not a big fan of those sorts of changes which take review bandwidth for no real reason, please plan accordingly such changes in the future to minimize review bandwidth for us (ensure it's draft, ping maintainers only when it's really ready), there have been a bunch of changes and I have been personally pinged because we were confused and you don't describe your intents.

Please add your reformatting commit to https://github.com/NixOS/nixpkgs/blob/master/.git-blame-ignore-revs to minimize disruption in the `git blame` output.

Please rename the PR title accordingly to your changes.

**Comment by RaitoBezarius** [comment]
The line breaking commit should go into the ignoring list too for the blame.

**Comment by AndersonTorres** [comment]
Not yet. I'll split mass-rebuilding change.

File: pkgs/os-specific/linux/systemd/default.nix [hunk 3]
```diff
@@ -68,16 +68,17 @@
...
   # Only libbpf should be a runtime dependency.
   # Note: llvmPackages is explicitly taken from buildPackages instead of relying
   # on splicing. Splicing will evaluate the adjacent (pkgsHostTarget) llvmPackages
   # which is sometimes problematic: llvmPackages.clang looks at targetPackages.stdenv.cc
-  # which, in the unfortunate case of pkgsCross.ghcjs, `throw`s. If we explicitly
-  # take buildPackages.llvmPackages, this is no problem because
-  # `buildPackages.targetPackages.stdenv.cc == stdenv.cc` relative to us. Working
-  # around this is important, because systemd is in the dependency closure of
-  # GHC via emscripten and jdk.
+  # which, in the unfortunate case of pkgsCross.ghcjs, `throw`s. If we
```
[the remark this answers is not in the document]
Comment (flokli) [reply pkgs/os-specific/linux/systemd/default.nix:77]: This is out of scope for this PR. Please open an RFC.

[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/os-specific/linux/systemd/default.nix:77]: I won't be doing that since the strain the RFC process is putting on people is unbearable for me.

**Pull request merged**